lib/ui/platform_dispatcher.dart

```diff
@@ -1642,31 +1642,53 @@ class FrameTiming {
 /// States that an application can be in.
 ///
 /// The values below describe notifications from the operating system.
-/// Applications should not expect to always receive all possible
-/// notifications. For example, if the users pulls out the battery from the
-/// device, no notification will be sent before the application is suddenly
-/// terminated, along with the rest of the operating system.
+/// Applications should not expect to always receive all possible notifications.
+/// For example, if the users pulls out the battery from the device, no
+/// notification will be sent before the application is suddenly terminated,
+/// along with the rest of the operating system.
+///
+/// For historical and name collision reasons, Flutter's application state names
+/// do not correspond 1:1 with the state names on all platforms. On Android, for
+/// instance, when the OS calls
+/// [`Activity.onPause`](https://developer.android.com/reference/android/app/Activity#onPause()),
+/// Flutter will enter the `inactive` state, but when Android calls
+/// [`Activity.onStop`](https://developer.android.com/reference/android/app/Activity#onStop()),
+/// Flutter enters the `paused` state. See the individual state documenation for
+/// descriptions of what they mean on each platform.
 ///
 /// See also:
 ///
-///  * [WidgetsBindingObserver], for a mechanism to observe the lifecycle state
-///    from the widgets layer.
+/// * [WidgetsBindingObserver], for a mechanism to observe the lifecycle state
+///   from the widgets layer.
 enum AppLifecycleState {
-  /// The application is visible and responding to user input.
+  /// The application is visible and responsive to user input.
+  ///
+  /// On Android, this state corresponds to an app or the Flutter host view
+  /// having focus while in Android's "resumed" state. It is possible for the
+  /// Flutter app to be in the [inactive] state while still being in Android's
+  /// "resumed" state if the app has lost focus, but hasn't had
```

Comment:
this is first paragraph where the word 'focus' is used in this file, and you're already saying here that it !=== to "resumed" which is a better known concept. Let's be very technically specific on what "focus" means and add links

Reply:
Okay, PTAL, I think I addressed this.